### PR TITLE
Add Peanut script

### PR DESCRIPTION
**Pull request by remz1337**
## Description

This PR adds script for the Peanut app (web dashboard for NUT)
https://github.com/Brandawg93/PeaNUT

## Type of change

- [X] New Script (Develop a new script or set of scripts that are fully functional and thoroughly tested)
- [X] I have performed a self-review of my code, adhering to established codebase patterns and conventions.

### CODE REVIEW

File: ct/peanut.sh [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  SD=""
+  NS=""
+  MAC=""
+  VLAN=""
+  SSH="no"
+  VERB="no"
+  echo_default
+}
+
+function update_script() {
```
Comment (tteck) [inline ct/peanut.sh:56]: Refactor. You never build the application after download. ??
```
function update_script() {
  header_info
  if [[ ! -f /etc/systemd/system/peanut.service ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
  RELEASE=$(curl -sL https://api.github.com/repos/Brandawg93/PeaNUT/releases/latest | grep '"tag_name":' | cut -d'"' -f4)
  if [[ ! -f /opt/${APP}_version.txt ]] || [[ "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]]; then
    msg_info "Updating $APP to ${RELEASE}"
    systemctl stop peanut
    wget -qO peanut.tar.gz https://api.github.com/repos/Brandawg93/PeaNUT/tarball/${RELEASE}
    tar -xzf peanut.tar.gz -C /opt/peanut --strip-components 1
    rm peanut.tar.gz
    cd /opt/peanut
    pnpm i &>/dev/null
    pnpm run build &>/dev/null
    systemctl start peanut
    echo "${RELEASE}" >/opt/${APP}_version.txt
    msg_ok "Updated $APP to ${RELEASE}"
  else
    msg_ok "No update required. ${APP} is already at ${RELEASE}"
  fi
  exit
}
```

File: ct/peanut.sh [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+description
+
+msg_info "Setting Container to Normal Resources"
+pct set $CTID -memory 512
+pct set $CTID -cores 1
+msg_ok "Set Container to Normal Resources"
+msg_ok "Completed Successfully!\n"
+echo -e "${APP} should be reachable by going to the following URL.
+         ${BL}http://${IP}:3000${CL} \n
+		 ${YW}Don't forget to update the NUT configuration!${CL} \n"
```
Comment (tteck) [inline ct/peanut.sh]: This goes in the instructions, along with the config path. Remove

File: install/peanut-install.sh [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+catch_errors
+setting_up_container
+network_check
+update_os
+
+msg_info "Installing Dependencies"
+$STD apt-get install -y curl
+$STD apt-get install -y sudo
+$STD apt-get install -y mc
+$STD apt-get install -y wget
```
Comment (tteck) [inline install/peanut-install.sh]: `wget` comes preinstalled on Debian

File: install/peanut-install.sh [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_20.x nodistro main" >/etc/apt/sources.list.d/nodesource.list
+$STD apt-get update
+$STD apt-get install -y nodejs
+msg_ok "Installed Node.js"
+
+msg_info "Installing NUT"
+$STD apt-get install -y nut-client
+msg_ok "Installed NUT"
+
+msg_info "Installing Peanut"
```
Comment (tteck) [inline install/peanut-install.sh:36]: More transparent (link). You also use `pnpm` to install, but used `npm` to build. ??
```
RELEASE=$(curl -sL https://api.github.com/repos/Brandawg93/PeaNUT/releases/latest | grep '"tag_name":' | cut -d'"' -f4)
wget -qO peanut.tar.gz https://api.github.com/repos/Brandawg93/PeaNUT/tarball/${RELEASE}
mkdir -p /opt/peanut
tar -xzf peanut.tar.gz -C /opt/peanut --strip-components 1
rm peanut.tar.gz
cd /opt/peanut
$STD npm install -g pnpm
$STD pnpm i
$STD pnpm run build
cp -r .next/static .next/standalone/.next/
```

**Comment by remz1337** [comment]
Thanks for the feedback. Addressed and tested successfully

**Comment by OddSquirrel** [comment]
Working just fine, but maybe a quick hint on the website would be in order that additional settings need to be made in `/etc/systemd/system/peanut.service`. Otherwise, folks might end up being confused when they only get an error message after visiting the proposed URL. 😎

**Comment by OddSquirrel** [comment]
> Thanks for the feedback. Addressed and tested successfully

@remz1337 Did you ever manage to see a persistent graph? For some reason, that window is always blank when I open the website—just like with the original Docker version, which is the reason I switched to this LXC in the first place. 🙈

**Comment by remz1337** [comment]
No, the data is not persisted. I think there's already a discussion about that on the official repo

**Comment by remz1337** [comment]
> Working just fine, but maybe a quick hint on the website would be in order that additional settings need to be made in `/etc/systemd/system/peanut.service`. Otherwise, folks might end up being confused when they only get an error message after visiting the proposed URL. 😎

@tteck One would need to configure NUT (files in `/etc/nut`) and update the peanut service environment variables (in `/etc/systemd/system/peanut.service`). Can I help you update the installation instruction on your website?

**Pull request merged**